sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -295,6 +295,28 @@ class StateStoreSuite extends SparkFunSuite with BeforeAndAfter with PrivateMeth
     provider.getStore(0).commit()
   }
 
+  test("SPARK-19779: A tmp file of delta file should not be reserved on HDFS " +
```

Comment:
Instead of adding a new test, I prefer to just add several lines to the above `SPARK-19677: Committing a delta file atop an existing one should not fail on HDFS`. E.g.

```
  test("SPARK-19677: Committing a delta file atop an existing one should not fail on HDFS") {
    val conf = new Configuration()
    conf.set("fs.fake.impl", classOf[RenameLikeHDFSFileSystem].getName)
    conf.set("fs.default.name", "fake:///")

    val provider = newStoreProvider(hadoopConf = conf)
    provider.getStore(0).commit()
    provider.getStore(0).commit()

    // Verify we don't leak temp files
    val tempFiles = FileUtils.listFiles(new File(provider.id.checkpointLocation), null, true)
      .asScala.filter(_.getName.contains("temp-"))
    assert(tempFiles.isEmpty)
  }
```